test-framework/kubernetes-client/src/main/java/io/quarkus/test/kubernetes/client/AbstractKubernetesFixturesTestResource.java

```diff
@@ -0,0 +1,55 @@
+package io.quarkus.test.kubernetes.client;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+
+import io.fabric8.kubernetes.api.model.HasMetadata;
+
+public abstract class AbstractKubernetesFixturesTestResource extends
+        AbstractNamespaceCreatingTestResource {
+    private List<HasMetadata> resourceFixtures = Collections.emptyList();
+    private int waitAtMostSecondsForFixturesReadiness;
+    private boolean createNSIfNeeded;
+
+    @Override
+    protected Map<String, String> doStart() {
+        final var resources = client().resourceList(resourceFixtures);
+        resources.accept(HasMetadata.class,
+                hasMetadata -> logger().info("Creating '{}' {} in namespace '{}'",
+                        hasMetadata.getMetadata().getName(),
+                        hasMetadata.getKind(), namespace()));
+        resources.createOrReplace();
+        resources.waitUntilReady(waitAtMostSecondsForFixturesReadiness, TimeUnit.SECONDS);
```

Comment:
In some iterations of the client this particular method was throwing an error if the target resource is not waitable. 
So, in similar functionality I ended up doing some things more manually: You may or may not find this handy: https://github.com/dekorateio/dekorate/blob/main/testing/kubernetes-junit/src/main/java/io/dekorate/testing/kubernetes/KubernetesExtension.java#L185

Reply:
As far as I'm aware that's not the case anymore, though a warning is logged. Maybe a better option would be to have a marker interface to properly detect resources that can be waited on?